### ISSUE TITLE
ch07 sec01 - 7.6.2 loss

### ISSUE THREAD

**Issue by d-kleine** [rasbt/reasoning-from-scratch#220]
In section 7.6.2, I have also wondered why `loss`and `kl_loss` are both zero when taking the metrics from *7_6_plus_format_reward_metrics.csv*. When plotting, you can see that both `loss`and `kl_loss` remain 0 across all trainings steps.

**Comment by rasbt** [comment]
Ah yes, thanks for the catch here. I disabled KL loss there (it's the default in the script), and the 0 loss is an implementation artifact. I am adding a note to the notebook and chapter:

> **Note on `loss` and `kl_loss`**
>
> - This experiment uses `7_6_plus_format_reward.py` with the default settings `--kl_coeff 0.0` and `--inner_epochs 1`
> - Because `--kl_coeff` is set to `0.0`, the KL penalty is disabled, so the logged `kl_loss` remains `0` throughout the run
>
> - If you want to enable KL regularization with the same settings used in Section 7.5, run:
>
> ```bash
> uv run 7_6_plus_format_reward.py \
>     --steps 500 \
>     --max_new_tokens 1024 \
>     --kl_coeff 0.001 \
>     --inner_epochs 2
> ```
>
> - The logged `loss` is also `0` with the default 7.6 settings
> - This happens because the run uses only one inner update (`--inner_epochs 1`) and compares the model against a copy of itself from the same step
> - As a result, the policy ratio starts at `1`, and since the advantages are normalized to have mean `0`, the scalar policy loss evaluates to `0`
>
> - So in this case, `loss = 0` is a computation artifact that is a consequence of the default settings
> -  I.e., it mainly reflects that the run uses a single inner epoch and no KL term, which makes the logged scalar loss uninformative
> - The more useful quantities to monitor here are metrics such as `reward_avg`, `format_reward_avg`, `adv_std`, and `entropy_avg`

**Comment by d-kleine** [comment]
Great, that makes it clear! Thank you! 👍🏻 🙂
